MSAL/src/MSALPublicClientApplication.m

```diff
@@ -114,11 +107,46 @@ + (void)load
     MSIDNotifications.webAuthDidReceiveResponseFromBrokerNotificationName = MSALWebAuthDidReceieveResponseFromBroker;
 }
 
+#pragma mark - Properties
+- (BOOL)validateAuthority { return YES; }
+- (void)setValidateAuthority:(BOOL)validateAuthority
+{
+    if ([self.configuration.authority isKindOfClass:MSALADFSAuthority.class])
+    {
+        ((MSALADFSAuthority *)self.configuration.authority).validateAuthority = validateAuthority;
+    }
+    else
+    {
+        MSID_LOG_WARN(nil, @"Authority validation is only available for authority targetting AD FS.");
+    }
+}
+
+- (MSALAuthority *)authority { return self.configuration.authority; }
+- (NSString *)clientId { return self.configuration.clientId; }
+- (MSALRedirectUri *)redirectUri { return self.configuration.verifiedRedirectUri; }
+
+- (NSDictionary<NSString *,NSString *> *)sliceParameters { return self.configuration.sliceConfig.sliceDictionary; }
+- (void)setSliceParameters:(NSDictionary<NSString *,NSString *> *)sliceParameters
+{
+    self.configuration.sliceConfig = [MSALSliceConfig configWithSlice:sliceParameters[@"slice"] dc:sliceParameters[@"dc"]];
+}
+
+- (MSALWebviewType)webviewType { return MSALGlobalConfig.defaultWebviewType; }
+- (void)setWebviewType:(MSALWebviewType)webviewType { MSALGlobalConfig.defaultWebviewType = webviewType; }
+
+#if TARGET_OS_IPHONE
+- (NSString *)keychainGroup { return MSALGlobalConfig.cacheConfig.keychainSharingGroup; }
+- (void)setKeychainGroup:(NSString * _Nullable)keychainGroup { MSALGlobalConfig.cacheConfig.keychainSharingGroup = keychainGroup; }
+#else
+- (NSString *)keychainGroup { return nil; }
+#endif
+
+#pragma mark - Initializers
 - (id)initWithClientId:(NSString *)clientId
                  error:(NSError * __autoreleasing *)error
 {
     return [self initWithClientId:clientId
-                    keychainGroup:self.defaultKeychainGroup
+                    keychainGroup:self.keychainGroup
```

Comment:
Will it always be set?

Reply:
This should be same as before. 

1. keychainGroup is specified with nil value:   [[NSBundle mainBundle] bundleIdentifier]
2. keychainGroup is specified with non-nil value, custom value is used.
3. keychainGroup is not specified: use default - com.microsoft.adalcache

I can make it a bit more clearer.

Reply:
In this call, it's 3 and the value comes from the default config.